android-pathview/src/main/java/com/eftimoff/androipathview/PathView.java

```diff
@@ -223,6 +261,44 @@ private void fillAfter(final Canvas canvas) {
         }
     }
 
+    /**
+     * If there is svg , the user called setFill(true).
+     *
+     * @param canvas Draw to this canvas.
+     */
+    private void fill(final Canvas canvas) {
+        if (svgResourceId != 0 && fill) {
+            svgUtils.drawSvgAfter(canvas, width, height);
+        }
+    }
+
+    /**
+     * If fillColor had value before then we replace untransparent pixels of bitmap by solid color
+     *
+     * @param bitmap Draw to this canvas.
+     */
+    private void applySolidColor(final Bitmap bitmap) {
+        if(fill && fillColor!=Color.argb(0,0,0,0) )
+            if (bitmap != null) {
```

Comment:
I'm not sure , but as I understood I think another way is applying color in SVG parsing level and that needs accessing and manipulating SVG parser source code. And in PathView project I think the only way is applying the color in canvas drawing level by changing pixels .
Maybe I'm wrong , then I will be happy to have your guide :)